### PR TITLE
fix error code in $? after a run command which was introduced by 6c8441

### PR DESCRIPTION
**Pull request by krimdomu**

### CODE REVIEW

File: lib/Rex/Interface/Exec/Local.pm [hunk 1]
```diff
@@ -119,6 +119,11 @@ sub exec {
 
   }
 
+  # we need to bitshift $? so that $? contains the right (and for all
+  # connection methods the same) exit code after a run()/i_run() call.
+  # this is for the user, so that he can query $? in his task.
+  $? >>= 8;
```
Comment (kablamo) [inline lib/Rex/Interface/Exec/Local.pm]: It seems like a bad idea to modify $?.  How will users no its not the usual unbitshifted value that Perl normally provides.  Why can't the user bitshift it 8 themselves if they want that value.

Comment (krimdomu) [inline lib/Rex/Interface/Exec/Local.pm]: Well this has historical reasons. The first connection method was Net::SSH2 which provided already bitshifted return values (afaik, which come directly from libssh2 library). So for all other connection methods i added this.

Also the normal shell behaviour (which admin knows) is to have "normal" return values and no need to bitshift them.

Comment (kablamo) [inline lib/Rex/Interface/Exec/Local.pm]: Just to clarify -- Net::SSH2 modified $?.

Comment (krimdomu) [inline lib/Rex/Interface/Exec/Local.pm]: No it doesn't modify it. The Net::SSH2::Channel object has a method `exit_value` (or something similar) which provides the "unbitshifted" return value. The overwirde of $? is done by rex.

**Pull request merged**